### PR TITLE
List directory contents non-recursively

### PR DESCRIPTION
**Pull request by zaak**
This PR allows for non-recursive content listing. The previous approach was always fetching all objects recursively (by prefix), which is really inefficient in case if some of the subdirectories contains a large number of files.

http://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysHierarchy.html
http://codereview.stackexchange.com/questions/6847/list-objects-in-a-amazon-s3-folder-without-also-listing-objects-in-sub-folders/6850#6850

### CODE REVIEW

File: src/AwsS3Adapter.php [hunk 1]
```diff
@@ -438,7 +448,7 @@ public function listContents($dirname = '', $recursive = false)
      */
     protected function normalizeResponse(array $object, $path = null)
     {
-        $result = ['path' => $path ?: $this->removePathPrefix($object['Key'])];
+        $result = ['path' => $path ?: $this->removePathPrefix(isset($object['Key']) ? $object['Key'] : $object['Prefix'])];
```
Comment (frankdejonge) [inline src/AwsS3Adapter.php:451]: What case is this addressing? Is this the response for the top level directory? Something else?

Comment (zaak) [inline src/AwsS3Adapter.php:451]: If content is listed non-recursively prefixes are listed without `Key` attribute. The path of the directory is stored as `Prefix`. Directory level doesn't matter here.

Comment (frankdejonge) [inline src/AwsS3Adapter.php:451]: I'm not sure I follow... If the directory is stored as Prefix, then all the files would be named the same, which wouldn't be correct. Or is the file path returned under the Prefix key?

Comment (zaak) [inline src/AwsS3Adapter.php:451]: Sorry for the delay. I've created a gist showing how the response from S3 looks like:
https://gist.github.com/zaak/6b0788dfbc1391849dbb, should be more clear.
I have a set of tests in our app that use Flysystem quite intensely, so I assume this change shouldn't break anything, as they all pass.

Comment (frankdejonge) [inline src/AwsS3Adapter.php:451]: And the best comment on a PR award goes tooooooooo ... @zaak! Seriously, thanks for this, this is exactly what I wanted to know.

Comment (frankdejonge) [inline src/AwsS3Adapter.php:451]: Btw, @zaak, could we use this for the V3 as well?

Comment (zaak) [inline src/AwsS3Adapter.php:451]: @frankdejonge I think yes, haven't checked it yet though.

Comment (frankdejonge) [inline src/AwsS3Adapter.php:451]: @zaak version 1.0.3 is tagged and should be available shortly.

Comment (zaak) [inline src/AwsS3Adapter.php:451]: @frankdejonge Awesome! Thanks! :)

**Pull request merged**